src/sentry/static/sentry/app/views/incidents/details/body.tsx

```diff
@@ -23,6 +26,36 @@ type Props = {
 } & RouteComponentProps<{incidentId: string; orgId: string}, {}>;
 
 export default class DetailsBody extends React.Component<Props> {
+  getDiscoverUrl(projects: Project[]) {
+    const {incident, params} = this.props;
+    const {orgId} = params;
+
+    if (!projects || !projects.length || !incident) {
+      return '';
+    }
+
+    const discoverQuery: NewQuery = {
+      id: undefined,
+      name: (incident && incident.title) || '',
+      fields: ['title', 'user', 'last_seen'],
+      widths: ['400', '200', '-1'],
+      orderby: '-last_seen',
+      query: (incident && incident.query) || '',
+      projects: projects
+        .filter(({slug}) => incident.projects.includes(slug))
+        .map(({id}) => Number(id)),
+      version: 2 as const,
+      range: '24h',
+    };
+
+    const discoverView = EventView.fromSavedQuery(discoverQuery);
+
+    return {
+      pathname: `/organizations/${orgId}/eventsv2/results/`,
+      query: discoverView.generateQueryStringObject(),
+    };
```

Comment:
I can't help but think that EventView should have a `generateResultsUrl()` method for this kind of scenario.

Reply:
Maybe. 👍